mino/minows/controller_test.go

```diff
@@ -0,0 +1,172 @@
+package minows
+
+import (
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/dela/cli/node"
+	"go.dedis.ch/dela/core/store/kv"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+func TestController_OnStart(t *testing.T) {
+	flags := new(mockFlags)
+	flags.On("String", "listen").Return("/ip4/0.0.0.0/tcp/8000/ws")
+	flags.On("String", "public").Return("/dns4/p2p-1.c4dt.dela.org/tcp/443/wss")
+
+	dir, err := os.MkdirTemp(os.TempDir(), "minogrpc")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+	db, err := kv.New(filepath.Join(dir, "test.db"))
```

Comment:
there's no purely in-memory db: `MemoryStorage` simply adds a memory-backed cache layer to the underlying file-based `DiskStorage`, so either one requires a physical file. same approach used in grpc.

Reply:
But this would work, no?

https://github.com/dedis/student_24_dela_libp2p/blob/main/testing/fake/store.go#L75

If you leave `err` and `errView` equal to `nil`, it should be a simple memory-based implementation of `kv.DB`.

Reply:
changed